{{cookiecutter.project_slug}}/Makefile

```diff
@@ -52,15 +53,17 @@ run: $(VENV)/pyvenv.cfg
 $(VENV)/pyvenv.cfg: pyproject.toml
 	# Create our Python 3 virtual environment
 	python3 -m venv env
-	$(VENV_BIN)/python -m pip install --upgrade pip
+	# NOTE(ekilmer): interrogate v1.5.0 needs setuptools when using Python 3.12+.
+	# This should be fixed when the next release is made
+	$(VENV_BIN)/python -m pip install --upgrade pip setuptools
 	$(VENV_BIN)/python -m pip install -e .[$(INSTALL_EXTRA)]
 
 .PHONY: lint
 lint: $(VENV)/pyvenv.cfg
 	. $(VENV_BIN)/activate && \
 		black --check $(ALL_PY_SRCS) && \
 		ruff $(ALL_PY_SRCS) && \
-		mypy $(PY_MODULE)
+		mypy
```

Comment:
Won't this end up typechecking the unit tests as well? I think that's why we needed `PY_MODULE` originally.

Reply:
This is one of the changes that _could_ be separated from this PR, sorry!

I've actually moved this to `pyproject.toml`:

```toml
[tool.mypy]
packages = "{{ cookiecutter.__project_import }}"
```

Also, running verbose mode with `mypy` shows no mention of the test file

```
env $ mypy -v

LOG:  Mypy Version:           1.6.1
LOG:  Config File:            pyproject.toml
LOG:  Configured Executable:  /private/tmp/work/python-project/env/bin/python
LOG:  Current Executable:     /private/tmp/work/python-project/env/bin/python
LOG:  Cache Dir:              .mypy_cache
LOG:  Compiled:               True
LOG:  Exclude:                []
LOG:  Found source:           BuildSource(path='src/python_project/__init__.py', module='python_project', has_text=False, base_dir=None, followed=False)
LOG:  Found source:           BuildSource(path='src/python_project/__main__.py', module='python_project.__main__', has_text=False, base_dir=None, followed=False)
LOG:  Found source:           BuildSource(path='src/python_project/_cli.py', module='python_project._cli', has_text=False, base_dir=None, followed=False)
LOG:  Metadata fresh for python_project: file src/python_project/__init__.py
LOG:  Metadata fresh for python_project.__main__: file src/python_project/__main__.py
LOG:  Metadata fresh for python_project._cli: file src/python_project/_cli.py
LOG:  Metadata fresh for builtins: file /private/tmp/work/python-project/env/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi
[...]
```

Reply:
Thinking about this more... I fixed a ruff error due to missing type hint in the test file, but since we don't want to run mypy over test files, we should probably revert that change and make ruff ignore the type hint check. https://github.com/trailofbits/cookiecutter-python/pull/5/files#diff-3c44fa2de190f87d81dff86464c50c96270f330e48f216cbdb6a4ef35c1d9c07